internal/store/driver.go

```diff
@@ -568,7 +701,74 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
 		edge.Spec.Metadata = d.customMetadata
 		edgeMap[domain.Spec.Domain] = edge
 	}
+	d.calculateHTTPSEdgesFromIngress(edgeMap)
+
+	if d.gatewayEnabled {
+
+		httproutes := d.store.ListHTTPRoutes()
+		gateways := d.store.ListGateways()
+		for _, gtw := range gateways {
+			var gatewayDomains []string
+			for _, listener := range gtw.Spec.Listeners {
+				if listener.Hostname == nil {
+					continue
+				}
+				gatewayDomains = append(gatewayDomains, string(*listener.Hostname))
+			}
+			if len(gatewayDomains) <= 0 {
+				continue
+			}
+			for _, httproute := range httproutes {
+				var routeDomains []string
+				for _, parent := range httproute.Spec.ParentRefs {
+					if string(parent.Name) != gtw.Name {
+						continue
+					}
+					var domainOverlap []string
+					for _, hostname := range httproute.Spec.Hostnames {
+						for _, parentDomain := range gatewayDomains {
+							if parentDomain == string(hostname) {
```

Comment:
I think you're correct, and I'll add that check, but I've avoided doing so as a convenience, since typing something like this:
```yaml
  listeners:
    - protocol: HTTPS
      hostname: <your domain>
      port: 443
      tls: 
        mode: Terminate
        certificateRefs:
        - name: <some placeholder for the cert ngrok creates>
      name: <your listener name>

```
Gets tedious

Reply:
skipped here: https://github.com/ngrok/kubernetes-ingress-controller/pull/343/commits/0ecef9755256f2bd5402e36dfad779e410106d4a